### ISSUE TITLE
Deploy to staging

### ISSUE THREAD

**Issue by DQGriffin** [RingCentral-Pro-Services/Croissant.js#3]
Please deploy the latest changes to staging. Required environment variables are listed below.

 - RC_PLATFORM_URL = https://platform.ringcentral.com
 - RC_CLIENT_ID = Croissant App Client ID
 - RC_CLIENT_SECRET = Croissant App Client Secret
 - RC_REDIRECT_URI = Croissant url followed by /oauth2callback
 - REACT_APP_CLIENT_ID = Croissant App Client ID
 - REACT_APP_AUTH_BASE = https://platform.ringcentral.com/restapi/oauth/authorize?response_type=code
 - REACT_APP_AUTH_REDIRECT = Croissant url followed by /oauth2callback

**Comment by Brutalbeard** [comment]
@DQGriffin for the app id/secret, where we gettin those from?

**Comment by DQGriffin** [comment]
@Brutalbeard I created the app in my test account - UID: 62811763028

**Comment by Brutalbeard** [comment]
@DQGriffin deployed to staging. test is out and let me know when we need to go to prod

staging-croissant.ps.ringcentral.com

**Comment by DQGriffin** [comment]
@Brutalbeard It looks like the environment variables below are undefined. Is this related to the frontend not being in the root directory?

RC_REDIRECT_URI
REACT_APP_CLIENT_ID
REACT_APP_AUTH_BASE
REACT_APP_AUTH_REDIRECT

**Comment by Brutalbeard** [comment]
Possibly? @pachori-ujjwal got any ideas? Maybe should be in the docker build file?

**Comment by pachori-ujjwal** [comment]
Yeah @Brutalbeard the issue is caused by build being made in Dockerfile. React is not getting the `REACT_APP_*` environment variables because those are not available to the `npm run build` command while the image is being built. Only `ARGV` variables are available at time of image building and `ENV` variables are set and available when container starts.

There are two possible solutions here:

1. Move the build to a script that is executed at time of container start.
2. Convert these environment variables to argument variables and pass them in the `docker build` command and then pass those argument variables as environment variables to the `npm run build` command in Dockerfile.

**Comment by Brutalbeard** [comment]
Sounds like option 1 is a little more preferrable :-)

**Comment by DQGriffin** [comment]
Okay, @Brutalbeard I've updated the docker file to build when the container starts. Should be good to go™

**Comment by Brutalbeard** [comment]
Deployed to staging! Time to go test @DQGriffin
